apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx

```diff
@@ -207,4 +207,36 @@ export default class ChooseYourProgram extends LabeledFormComponent {
 
     return changes;
   }
+
+  /**
+   * @override
+   */
+  static getErrorMessages(data) {
+    let formatErrors = {};
+    if (
+      data.csHowManyMinutes &&
+      (data.csHowManyMinutes < 1 || data.csHowManyMinutes > 480)
+    ) {
+      formatErrors.csHowManyMinutes =
+        'Class section minutes per day must be positive number from 1-480 inclusive';
+    }
+
+    if (
+      data.csHowManyDaysPerWeek &&
+      (data.csHowManyDaysPerWeek < 1 || data.csHowManyDaysPerWeek > 7)
+    ) {
+      formatErrors.csHowManyDaysPerWeek =
+        'Class section days per week must be positive number between 1-7 inclusive';
+    }
+
+    if (
+      data.csHowManyWeeksPerYear &&
+      (data.csHowManyWeeksPerYear < 1 || data.csHowManyWeeksPerYear > 52)
+    ) {
+      formatErrors.csHowManyWeeksPerYear =
+        'Class section weeks per year must be positive number from 1-52 inclusive';
```

Comment:
Is this error text user-facing?

Reply:
Yes. Do you want to change it?

IMO, a better user experience could be that we show the expected values before users try to answer a question. That would eliminate a lot of confusion and mistakes. However, that is another discussion.

Reply:
Agreed on UX—primary motivation for now is to keep clean up the data. Let's tweak the text to: "Class section weeks per year must be between 1 and 52". Just going for something slightly more friendly. I'll leave a couple other comments with similar requests

Reply:
All fixed.